src/App.tsx

```diff
@@ -1,72 +1,130 @@
-import React from 'react';
+import React, { useMemo, useRef, useState } from 'react';
+import debounce from 'lodash.debounce';
 import './App.scss';
 import { peopleFromServer } from './data/people';
+import { Person } from './types/Person';
 
-export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+interface AppProps {
+  debounceDelay?: number;
+  onSelected?: (person: Person | null) => void;
+}
+
+export const App: React.FC<AppProps> = ({
+  debounceDelay = 300,
+  onSelected,
+}) => {
+  const [query, setQuery] = useState('');
+  const [appliedQuery, setAppliedQuery] = useState('');
+  const [selected, setSelected] = useState<Person | null>(null);
+  const [isDropdownVisible, setIsDropdownVisible] = useState(false);
+
+  const debouncedApplyQuery = useRef(
+    debounce((value: string) => {
+      setAppliedQuery(value);
+    }, debounceDelay),
+  ).current;
+
+  const filteredPeople = useMemo(() => {
+    if (!appliedQuery) {
+      return peopleFromServer;
+    }
+
+    return peopleFromServer.filter(person =>
+      person.name.toLowerCase().includes(appliedQuery.trim().toLowerCase()),
```

Comment:
The filtering logic should be adjusted to prevent filtering when only spaces are entered. Consider checking if `appliedQuery.trim()` is not empty before applying the filter. This aligns with the checklist requirement: "make sure that filter won't be called if user entered spaces only".